src/evict/evict_lru.c

```diff
@@ -2458,18 +2459,21 @@ __evict_page(WT_SESSION_IMPL *session, bool is_server)
     time_start = 0;
 
     flags = 0;
+    cache_eviction_app_dirty_inc = 0;
 
     /*
      * An internal session flags either the server itself or an eviction worker thread.
      */
     if (is_server)
-        WT_STAT_CONN_INCR(session, cache_eviction_server_evicting);
+        WT_STAT_CONN_INCR(session, cache_eviction_server_evict_attempt);
     else if (F_ISSET(session, WT_SESSION_INTERNAL))
-        WT_STAT_CONN_INCR(session, cache_eviction_worker_evicting);
+        WT_STAT_CONN_INCR(session, cache_eviction_worker_evict_attempt);
     else {
-        if (__wt_page_is_modified(ref->page))
-            WT_STAT_CONN_INCR(session, cache_eviction_app_dirty);
-        WT_STAT_CONN_INCR(session, cache_eviction_app);
+        if (__wt_page_is_modified(ref->page)) {
+            cache_eviction_app_dirty_inc = 1;
```

Comment:
What is the point of the `cache_eviction_app_dirty_inc` var?

Reply:
Oh I see, I'd use a bool here instead.

Reply:
Processors can lose a lot of time in branch misprediction. By writing it this way I eliminated the need to conditionally branch and hence the possibility of branch misprediction.

Reply:
I'm not sure what the right answer is here but it's an interesting conversation to have. By using the int instead of a bool we are guaranteed to have the following code run: 
```
#define WT_STAT_INCRV_BASE(session, stat, fld, value) \
    do {                                              \
        if (WT_STAT_ENABLED(session))                 \
            (stat)->fld += (int64_t)(value);          \
    } while (0)
```
This itself has an if branch but this if branch is on a pointer load of `session->stat_flags` which is may not be in the CPU cache, although probably would be given the previous usages earlier in the function. Assuming we take that branch we have an addition and a store: (stat)->fld += (int64_t)(value); which would be probably be a RMW cycle. I think if the value being added is zero which it sometimes will be that may be optimized out either by the CPU of the compiler somehow. 

So the CPU could mispredict the added branch anyway. But we also add a load from the session structure to determine if we take that branch. 

Now if we consider the boolean option which is:
```
            if (app_dirty) { 
                 WT_STAT_CONN_INCR(session, cache_eviction_app_dirty_fail);
            }
            WT_STAT_CONN_INCR(session, cache_eviction_app_fail);
```
We do a comparison on a register holding the boolean value which will be faster but add an additional potential mispredicted branch.

I think without the appropriate profiling there's no way to say for sure which is best but I'd lean toward the register comparison as it meas we won't do subsequent loads and stores. The additional benefit of this path is that we use a boolean and an if statement which I think for new readers of the code will be more obvious than the integer. This is because the error path ifs will match the same ifs as the setup path.

Given that, I'd say readability trumps micro-optimization.

Reply:
I will change it to bool. I didn't look the expansion of WT_STAT_CONN_INCRV_BASE(). I thought it would just be an atomic increment. Since it has a conditional branch, I am changing to a bool.